src/test/scala/de/frosner/ddq/core/CheckTest.scala

```diff
@@ -489,10 +493,31 @@ class CheckTest extends FlatSpec with Matchers with BeforeAndAfterEach with Befo
     val ref = makeIntegersDf(List(1, 3, 100), List(1, 500, 100))
     val check = Check(base).isJoinableWith(ref, "column1" -> "column1", "column3" -> "column2")
     val constraint = check.constraints.head
-    val result = ConstraintSuccess("Columns column1->column1, column3->column2 can be used for joining (one distinct row match)")
+    val result = ConstraintSuccess("Columns column1->column1, column3->column2 can be used for joining (" +
+      "join columns cardinality in base table: 2, " +
+      "join columns cardinality after joining: 1 (50%)")
     check.run().constraintResults shouldBe Map(constraint -> result)
   }
 
+  it should "compute the matched keys in a non-commutative way" in {
+    val base = makeIntegerDf(List(1, 1, 1, 1, 1, 1, 1, 1, 1, 2))
+    val ref = makeIntegerDf(List(1))
+
+    val check1 = Check(base).isJoinableWith(ref, "column" -> "column")
+    val constraint1 = check1.constraints.head
+    val result1 = ConstraintSuccess("Column column->column can be used for joining (" +
+      "join columns cardinality in base table: 2, " +
+      "join columns cardinality after joining: 1 (50%)")
+    check1.run().constraintResults shouldBe Map(constraint1 -> result1)
+
+    val check2 = Check(ref).isJoinableWith(base, "column" -> "column")
+    val constraint2 = check2.constraints.head
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Usage of head on collections.](https://www.codacy.com/app/frank_7/drunken-data-quality/file/1998406483/issues/source?bid=3033737&fileBranchId=3048638#l514)